src/Microsoft.AspNet.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -24,22 +24,36 @@ public static IApplicationBuilder UseMiddleware(this IApplicationBuilder builder
             {
                 var instance = ActivatorUtilities.CreateInstance(builder.ApplicationServices, middleware, new[] { next }.Concat(args).ToArray());
                 var methodinfo = middleware.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public);
+                if (methodinfo == null)
```

Comment:
If there is more than one Invoke method GetMethod will throw an AmbigousMatchException. Are you OK with that or do you want to provide a more detailed error? You could call `GetMethods` and see how many you get back.